tfx/components/example_gen/utils.py

```diff
@@ -284,9 +326,26 @@ def calculate_splits_fingerprint_and_span(
       split.pattern = split.pattern.replace(SPAN_SPEC, select_span)
     if select_span is None:
       select_span = '0'
+
+    # Given previous span, find most recent version
+    if VERSION_SPEC in split.pattern:
+      latest_version = _retrieve_latest_version_from_span(input_base_uri,
+                                                          split,
+                                                          select_span)
+      logging.info('latest version = %s', latest_version)
+      if select_version is None:
+        select_version = latest_version
+      if select_version != latest_version:
+        raise ValueError(
```

Comment:
we should have the same for span, cause the structure should be span/version/split, thus all split of the same artifact should have same span and version

Reply:
i think Span logic should already be fine (unless it needs changes), since we check if our calculated `latest_span` is the same as the current `select_span`

https://github.com/18jeffreyma/tfx/blob/d7482f2502f4bc2472944e4cd1e40a7a7fc41b28/tfx/components/example_gen/utils.py#L280